x-pack/plugins/task_manager/server/config.ts

```diff
@@ -204,6 +204,7 @@ export const configSchema = schema.object(
     }),
     claim_strategy: schema.string({ defaultValue: CLAIM_STRATEGY_UPDATE_BY_QUERY }),
     request_timeouts: requestTimeoutsConfig,
+    auto_calculate_default_ech_capacity: schema.boolean({ defaultValue: false }),
```

Comment:
Do we need to add to the docker allowlist and the cloud allowlist?

Reply:
Hmm, good point. The original thinking was to revert this PR by 8.18 and ensure we're happy with the `HEAP_TO_CAPACITY_MAP` config based on production experiments, and use `xpack.task_manager.capacity` as the opt-out strategy. But I can see where we could use this as an opt-out mechanism as well. I'll take note to think it through, I'll add it to the dockerfile anyway in this PR https://github.com/elastic/kibana/pull/195390/commits/1deaff29d162d546dde5a7835c75e4b175a7a38f, leaving the cloud allow list that we'll need to add if ever we continue with this.

Reply:
Without it in the cloud allow list, it will only be possible to set it with the operator overrides capability.  That should be fine if we only need to deal with a few cases.

Also to keep in mind, I believe the cloud allow list stuff is only updated on releases, but not sure.  Meaning we may need to wait for a point release to wait for it to go into effect.

Reply:
++ I can expand a bit into the two options in 8.18 to rollback the auto calculated capacity:
1. Customer sets `xpack.task_manager.capacity` to an explicit value, which will take precedence over the calculated default value. In this case, we can make them set `10` or something else.
2. If we keep the feature flag, customers can set `xpack.task_manager. auto_calculate_default_ech_capacity` to `false`, which means we'll default to `10` normal tasks until they specify otherwise via `xpack.task_manager.capacity`. It's pretty much the same as asking them to put a capacity of `10` but with the added benefit that we can re-opt them into auto calculating when removing the `auto_calculate_default_ech_capacity` setting (breaking change).

It feels like option 1 is ok and we can remove this new `auto_calculate_default_ech_capacity ` setting in 8.18 when we no longer need this functionality off by default.